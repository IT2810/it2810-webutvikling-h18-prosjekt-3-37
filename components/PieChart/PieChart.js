import React from 'react';
import { StyleSheet, View } from 'react-native';

import { VictoryPie } from "victory-native";
import { Content } from 'native-base';

export default class PieChart extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      completedTasks: this.props.completedTasks,
      totalTasks: this.props.totalTasks
    };
  }

  render() {
    return (
      <VictoryPie
        height={200}
        innerRadius={70}
        labelRadius={80}
        colorScale={["#44FF44", "#888888" ]}
        data={[
          { x: "Completed: " + this.state.completedTasks, y: (this.state.completedTasks / this.state.totalTasks)* 100 },
          { x: "Remaining: " + (this.state.totalTasks - this.state.completedTasks), y: 100 - ((this.state.completedTasks / this.state.totalTasks)* 100)}
        ]}
      />
    );

  }
}
