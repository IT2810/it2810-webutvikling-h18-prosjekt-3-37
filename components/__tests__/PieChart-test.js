import 'react-native';
import React from 'react'
import renderer from 'react-test-renderer'

import PieChart from '../PieChart/PieChart'



it('renders correctly', () => {
    jest.useFakeTimers();
    const tree = renderer.create(<PieChart />);
    expect(tree).toMatchSnapshot();
});