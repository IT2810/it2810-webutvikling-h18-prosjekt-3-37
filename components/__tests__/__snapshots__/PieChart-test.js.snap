// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="VictoryContainer"
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
  touchAction="box-none"
>
  <RNSVGSvgView
    align="xMidYMid"
    aria-labelledby="victory-container-1-title victory-container-1-desc"
    bbHeight="200"
    bbWidth="750"
    meetOrSlice={0}
    minX={0}
    minY={0}
    role="img"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        false,
        Object {
          "flex": 0,
          "height": 200,
          "width": 750,
        },
      ]
    }
    vbHeight={200}
    vbWidth={750}
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={Array []}
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    >
      <RNSVGPath
        d="M4.2862637970157365e-15,-70L3.061616997868383e-15,-50Z"
        fill={
          Array [
            0,
            0.26666666666666666,
            1,
            0.26666666666666666,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            375,
            100,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth="1"
        x={0}
        y={0}
      />
      <RNSVGPath
        d="M4.2862637970157365e-15,-70L3.061616997868383e-15,-50Z"
        fill={
          Array [
            0,
            0.5333333333333333,
            0.5333333333333333,
            0.5333333333333333,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            375,
            100,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth="1"
        x={0}
        y={0}
      />
      <RNSVGText
        content={null}
        deltaX={
          Array [
            "0",
          ]
        }
        deltaY={
          Array [
            "-2.0300000000000002",
          ]
        }
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        positionX={
          Array [
            "375",
          ]
        }
        positionY={
          Array [
            "20",
          ]
        }
        propList={Array []}
        rotate={Array []}
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth="1"
        x={0}
        y={0}
      >
        <RNSVGTSpan
          content="Completed: undefined"
          deltaX={
            Array [
              "0",
            ]
          }
          deltaY={Array []}
          fill={
            Array [
              0,
              0.1450980392156863,
              0.1450980392156863,
              0.1450980392156863,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={
            Object {
              "fontFamily": "Gill Sans",
              "fontSize": "14",
              "letterSpacing": "normal",
              "textAnchor": "middle",
            }
          }
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          positionX={
            Array [
              "375",
            ]
          }
          positionY={Array []}
          propList={
            Array [
              "fill",
            ]
          }
          rotate={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth="1"
          x={0}
          y={0}
        />
      </RNSVGText>
      <RNSVGText
        content={null}
        deltaX={
          Array [
            "0",
          ]
        }
        deltaY={
          Array [
            "-2.0300000000000002",
          ]
        }
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        positionX={
          Array [
            "375",
          ]
        }
        positionY={
          Array [
            "20",
          ]
        }
        propList={Array []}
        rotate={Array []}
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth="1"
        x={0}
        y={0}
      >
        <RNSVGTSpan
          content="Remaining: NaN"
          deltaX={
            Array [
              "0",
            ]
          }
          deltaY={Array []}
          fill={
            Array [
              0,
              0.1450980392156863,
              0.1450980392156863,
              0.1450980392156863,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={
            Object {
              "fontFamily": "Gill Sans",
              "fontSize": "14",
              "letterSpacing": "normal",
              "textAnchor": "middle",
            }
          }
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          positionX={
            Array [
              "375",
            ]
          }
          positionY={Array []}
          propList={
            Array [
              "fill",
            ]
          }
          rotate={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth="1"
          x={0}
          y={0}
        />
      </RNSVGText>
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "top": 0,
        "zIndex": 99,
      }
    }
    touchAction="box-none"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="200"
      bbWidth="750"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "height": "100%",
            "overflow": "visible",
            "width": "100%",
          },
          false,
          Object {
            "flex": 0,
            "height": 200,
            "width": 750,
          },
        ]
      }
      vbHeight={200}
      vbWidth={750}
    />
  </View>
</View>
`;
